examples/some-verifier/src/db.rs

```diff
@@ -0,0 +1,121 @@
+use some_verifier_lib::Platform;
+use tokio_postgres::{NoTls, Row};
+
+/// A social media platform that can be stored in the database.
+pub trait DbPlatform {
+    /// The `Platform` enum value of the platform.
+    const PLATFORM: Platform;
+}
+
+/// A trait that is implemented for the Platform enum to give some utility functons.
+trait DbName {
+    /// The name of the corresponding table.
+    fn table_name(&self) -> &'static str;
+    /// The name of the corresponding column.
+    fn column_name(&self) -> &'static str;
+}
+
+macro_rules! db_platform {
+    {$($name:ident { db_name = $db_name:literal $(,)? })* } => {
+        $(
+            pub struct $name;
+            impl DbPlatform for $name {
+                const PLATFORM: Platform = Platform::$name;
+            }
+        )*
+
+        impl DbName for Platform {
+            fn table_name(&self) -> &'static str {
+                match self {$(
+                    Platform::$name => $db_name,
+                )*}
+            }
+
+            fn column_name(&self) -> &'static str {
+                match self {$(
+                    Platform::$name => concat!($db_name, "_id"),
+                )*}
+            }
+        }
+
+        fn accounts_from_row(row: Row) -> Vec<Account> {
+            vec![ $( Account::from_row::<$name>(&row)),* ]
```

Comment:
It's a bit odd to return a singleton vector which I think this ends up being.

Reply:
It returns a vector with one `Account` per platform (so Telegram and Discord right now)